DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1

```diff
@@ -0,0 +1,168 @@
+######################################################################################
+# DSC Resource for IIS Server level MIME Type mappings
+######################################################################################
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+NoWebAdministrationModule=Please ensure that WebAdministration module is installed.
+AddingType=Adding MIMEType '{0}' for extension '{1}'
+RemovingType=Removing MIMEType '{0}' for extension '{1}'
+TypeExists=MIMEType '{0}' for extension '{1}' already exist
+TypeNotPresent=MIMEType '{0}' for extension '{1}' is not present as requested
+TypeStatusUnknown=MIMEType '{0}' for extension '{1}' is is an unknown status
+'@
+}
+
+######################################################################################
+# The Get-TargetResource cmdlet.
+# This function will get the Mime type for a file extension
+######################################################################################
+function Get-TargetResource
+{
+    [OutputType([Hashtable])]
+	param
+	(		
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$Extension,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [String]$MimeType,
+
+        [ValidateSet("Present", "Absent")]
+        [Parameter(Mandatory)]
+        [string]$Ensure
+	)
+	
+    # Check if WebAdministration module is present for IIS cmdlets
+    CheckIISPoshModule
+
+    $getTargetResourceResult = $null;
+
+    $mt = GetMapping $Extension $MimeType 
+
+    if ($mt -eq $null)
+    {
+        $ensureResult = "Absent";
```

Comment:
Even if we don't have the data we should be returning the EnsureResult gets returned or we should remove it. Right now $ensureResult is being assigned with no way to acuatlly use the result. The way I typically design my get-targetresources I would return:

	if ($mt -eq $null)
	{
		return @{
			Ensure = 'Absent'
			Extension = $null
			MimeType = $null
		}
	}
	else
	{
		return @{
			Ensure = 'Present'
			Extension = $mt.fileExtension
			MimeType = $mt.mimeType
		}
	}

Also I'd drop assigning $getTargetResourceResult to null as it's not really needed.